packages/slonik/src/dbFilters.ts

```diff
@@ -0,0 +1,114 @@
+import { QueryResultRow, TaggedTemplateLiteralInvocation, sql } from "slonik";
+
+import { FilterInput } from "./types";
+
+const applyFilter = (filter: FilterInput, tableName: string) => {
+  const key = filter.key;
+  const operator = filter.operator || "eq";
+  const not = filter.not || false;
+  let value: TaggedTemplateLiteralInvocation<QueryResultRow> | string =
+    filter.value;
+
+  const databaseField = sql.identifier([tableName, key]);
+  let clauseOperator;
+
+  switch (operator) {
+    case "ct":
+    case "sw":
+    case "ew": {
+      const valueString = {
+        ct: `%${value}%`, // contains
+        ew: `%${value}`, // ends with
+        sw: `${value}%`, // starts with
+      };
+
+      value = valueString[operator];
+      clauseOperator = not ? sql`NOT ILIKE` : sql`ILIKE`;
+      break;
+    }
+    case "eq":
+    default: {
+      clauseOperator = not ? sql`!=` : sql`=`;
+      break;
+    }
+    case "gt": {
+      clauseOperator = not ? sql`<` : sql`>`;
+      break;
+    }
+    case "gte": {
+      clauseOperator = not ? sql`<` : sql`>=`;
+      break;
+    }
+    case "lte": {
+      clauseOperator = not ? sql`>` : sql`<=`;
+      break;
+    }
+    case "lt": {
+      clauseOperator = not ? sql`>` : sql`<`;
+      break;
+    }
+    case "in": {
+      clauseOperator = not ? sql`NOT IN` : sql`IN`;
+      value = sql`(${sql.join(value.split(","), sql`, `)})`;
+      break;
+    }
+    case "bt": {
+      clauseOperator = not ? sql`NOT BETWEEN` : sql`BETWEEN`;
+      value = sql`${sql.join(value.split(","), sql` AND `)}`;
+      break;
+    }
+  }
+
+  return sql`${databaseField} ${clauseOperator} ${value}`;
+};
+
+const applyFiltersToQuery = (
+  filters: FilterInput,
+  tableName: string,
+  not = false
+) => {
+  const andFilter: TaggedTemplateLiteralInvocation<QueryResultRow>[] = [];
+  const orFilter: TaggedTemplateLiteralInvocation<QueryResultRow>[] = [];
+  let queryFilter;
+
+  const applyFilters = (
```

Comment:
So the `applyFiltersToQuery method creates and invokes a `applyFilters` method? Why not just run the code?